hawc/apps/common/exports.py

```diff
@@ -0,0 +1,108 @@
+import pandas as pd
+
+
+class Module:
+    def __init__(
+        self,
+        key_prefix: str = "",
+        query_prefix: str = "",
+        include: tuple | None = None,
+        exclude: tuple | None = None,
+    ):
+        # handle include, exclude, prefix?
+        self.key_prefix = key_prefix + "-" if key_prefix else key_prefix
+        self.query_prefix = query_prefix + "__" if query_prefix else query_prefix
+        self.include = (key_prefix + field for field in include) if include else tuple()
+        self.exclude = (key_prefix + field for field in exclude) if exclude else tuple()
+
+    @property
+    def value_map(self):
+        if hasattr(self, "_value_map"):
+            return self._value_map
+
+        value_map = self._get_value_map()
+        # add key prefix
+        if self.key_prefix:
+            value_map = {self.key_prefix + k: v for k, v in value_map.items()}
+        # add query prefix
+        if self.query_prefix:
+            value_map = {k: self.query_prefix + v for k, v in value_map.items()}
+        # handle any includes
+        if self.include:
+            value_map = {k: v for k, v in value_map.items() if k in self.include}
+        # handle any excludes
+        if self.exclude:
+            value_map = {k: v for k, v in value_map.items() if k not in self.exclude}
+
+        self._value_map = value_map
+        return self._value_map
+
+    @property
+    def annotation_map(self):
+        if hasattr(self, "_annotation_map"):
+            return self._annotation_map
+
+        annotation_map = self._get_annotation_map(self.query_prefix)
+        # add query prefix
+        if self.query_prefix:
+            annotation_map = {self.query_prefix + k: v for k, v in annotation_map.items()}
+        # handle any includes/excludes
+        if self.include or self.exclude:
+            annotation_map = {
+                k: v for k, v in annotation_map.items() if k in self.value_map.values()
+            }
+
+        self._annotation_map = annotation_map
+        return self._annotation_map
+
+    def _get_value_map(self):
```

Comment:
I think these are our public API for this class, so I'd remove the underscores. We should also add type annotations for the public methods too